### PR TITLE
Implement features for Nix language server (Nil)

### PR DESCRIPTION
**Pull request by psibi**
More details about the configuration here:
https://github.com/oxalica/nil/blob/main/docs/configuration.md

### CODE REVIEW

File: clients/lsp-nix.el [hunk 1]
```diff
@@ -54,9 +54,38 @@
...
+   ("nil.diagnostics.ignored" lsp-nix-nil-ignored-diagnostics)
+   ("nil.diagnostics.excludedFiles" lsp-nix-nil-exclude-files-diagnostic)))
+
 (lsp-register-client
  (make-lsp-client :new-connection (lsp-stdio-connection (lambda () lsp-nix-nil-server-path))
                   :major-modes '(nix-mode)
+                  :initialized-fn (lambda (workspace)
+                    (with-lsp-workspace workspace
+                      (lsp--set-configuration
+                       (lsp-configuration-section "nil"))))
```
Comment (yyoncho) [inline clients/lsp-nix.el:88]: you may add `:synchronize-sections '("nil")` as well.

File: clients/lsp-nix.el [hunk 2]
```diff
@@ -54,9 +54,38 @@
   :type 'string
   :package-version '(lsp-mode . "8.0.1"))
 
+(defcustom lsp-nix-nil-formatter nil
```
Comment (yyoncho) [inline clients/lsp-nix.el:57]: can you use defcustom-lsp?

Comment (psibi) [reply clients/lsp-nix.el]: Thanks, didn't know about it and it makes things much simplified.

File: clients/lsp-nix.el (hunk 1 above)
Comment (psibi) [reply clients/lsp-nix.el:86]: Sure, added it.

**Review by yyoncho — APPROVED** [review]

**Review by jcs090218 — APPROVED** [review]

**Pull request merged**